clients/fides-js/src/lib/cookie.ts

```diff
@@ -215,26 +211,33 @@ export const saveFidesCookie = (
   // eslint-disable-next-line no-param-reassign
   cookie.fides_meta.updatedAt = updatedAt;
 
-  // Write the cookie to the root domain
-  const rootDomain = window.location.hostname.split(".").slice(-2).join(".");
-
   let encodedCookie: string = JSON.stringify(cookie);
   if (base64Cookie) {
     encodedCookie = base64_encode(encodedCookie);
   }
 
-  setCookie(
-    CONSENT_COOKIE_NAME,
-    encodedCookie,
-    {
-      // An explicit path ensures this is always set to the entire domain.
-      path: "/",
-      // An explicit domain allows subdomains to access the cookie.
-      domain: rootDomain,
-      expires: CONSENT_COOKIE_MAX_AGE_DAYS,
-    },
-    CODEC
-  );
+  const hostname = window.location.hostname.split(".");
+  let topViableDomain = "";
+  for (let i = 1; i < hostname.length + 1; i += 1) {
+    // This loop guarantees to get the top-level hostname because that's the smallest one browsers will let you set cookies in. We test a given suffix for whether we are able to set cookies, if not we try the next suffix until we find the one that works.
+    topViableDomain = hostname.slice(-i).join(".");
```

Comment:
So that I understand, for something like `app.privacy.example.com` this loop will do:
1. `com`
2. `example.com`
3. `privacy.example.com`
4. `app.privacy.example.com`

?

Reply:
no, once the loop is able to set the cookie the loops stops with the break. In your example, the loop will do:

1. `com`
2. `example.com`

And now the cookie is set and will `break`